### PR TITLE
The Great Hat Overhaul

### PR DESCRIPTION
**Pull request by TheRealScarHomie**
## About The Pull Request

Implements custom made fallout accurate hats, masks and glasses, effectively annihilating every legacy clothing sprite in our codebase, we can now rest easy for we have the swag, head to mo-fuckin toe. (No custom inhands on these but thas bout it)

22 Helmets
image

48 Hats
image

20-ish Face/Eye Things
image

## List:
_`Helmets (full cover)  22 helmets

 Assault gas mask
 Assaultron helmet
 Brown flight helmet
 red flight helmet
 yellow flight helmet
 flight helmet
 banded helmet
 bladed helmet
 plated cowl
 cowl
 spiked junk helmet
 gas mask
 Inquisitor's cowl
 tactical wetsuit hood
 muffalo skull
 radstag skull
 sack hood
 sack hood with spikes
 sack hood with straps
 security helmet
 sentry bot helmet
 vault tec sec helmet
 sentry bot helmet

Hats 48 hats

 Captain's hat
 Baseball cap
 Faded visor	
 Fedoras	
 Bowler hat
 Cowboy hats
 Black prospector's hat 
 Batting helmets
 Chef hat
 Coast Guard hat	
 Dirty postman hat
 Police hat	
 Eyebot helmet
 Fisherman's hat
 Knit caps	
 Green hood
 Green rag hat
 Hard hat
 Hunter's hood
 Mayors Bicorn
 Mascot head
 Military cap	
 Medical goggles
 Militia hat
 Militia Beret
 Tricorn
 Mining helmet
 Newsboy cap
 Pirate hat
 Prospector's hat
 Sea captain's hat	
 Submariner/Communist hat
 Trilby hat
 Ushankas
 Welding helmet	
 Wool fisherman's cap
 Top Hat`_

back to clothing hibernation 🏏

### CODE REVIEW

File: mojave/items/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -1,11 +1,86 @@
...
-	desc = "A nice thick pair of goggles with a questionably attached band around it. Good for keeping debris out of your eyes, wether it be dust or viscera."
-	icon_state = "biker"
+/obj/item/clothing/glasses/ms13/black
+	name = "black glasses"
+	desc = "Snazzy glasses for the hip or near-blind."
+	icon_state = "black"
+
+/obj/item/clothing/glasses/ms13/cap
+	name = "bottle-cap glasses"
+	desc = "Crude soda caps construed into spectacles, they provide some sort of uh ... Quirkiness."
```
Comment (hyperioo) [inline mojave/items/clothing/glasses/glasses.dm:20]: Did you mean "transformed into"? `construed` means interpreted

File: mojave/items/clothing/glasses/glasses.dm [hunk 2]
```diff
@@ -1,11 +1,86 @@
...
+	flags_cover = GLASSESCOVERSEYES
+
+/obj/item/clothing/glasses/ms13/sunglasses
+	name = "sunglasses"
+	desc = "Strangely ancient technology used to help provide rudimentary eye cover from the harsh mojave sun."
+	icon_state = "sunglasses"
+
+/obj/item/clothing/glasses/ms13/welding
+	name = "welding goggles"
+	desc = "Arclight insured RobCo pre-war welding goggles, helps you keep your sight."
```
Comment (hyperioo) [inline mojave/items/clothing/glasses/glasses.dm:80]:
```suggestion
	desc = "Arclight insured RobCo pre-war welding goggles, help you keep your sight."
```
Goggles are plural

File: mojave/items/clothing/head/helmet.dm [hunk 3]
```diff
@@ -3,47 +3,182 @@
...
+	dynamic_hair_suffix = ""
+	dynamic_fhair_suffix = ""
 
-/obj/item/clothing/head/helmet/ms13/utilitycap/blue
-	name = "blue utility cap"
-	desc = "A pleasant blue utility cap. The brim is a bit floppy."
-	icon_state = "utility_nsoft"
+/obj/item/clothing/head/helmet/ms13/flight
+	name = "old flight helmet"
+	desc = "An old rusted flight helmet, hasnt seen use in a millenia."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:48]: A millenia? Not a century?

File: mojave/items/clothing/head/helmet.dm [hunk 4]
```diff
@@ -3,47 +3,182 @@
...
+	desc = "Health and safety at its finest."
+	icon_state = "hardhat"
+	inhand_icon_state = "vaultvanarmor"
+	armor = list(melee = 30, bullet = 0, laser = 5, energy = 10, bomb = 10, bio = 5, rad = 5, fire = 10, acid = 5)
+	dynamic_hair_suffix = ""
+	dynamic_fhair_suffix = ""
+
+/obj/item/clothing/head/hardhat/ms13/mining
+	name = "mining hardhat"
+	desc = "A sturdy helmet with a built-in nuclear powered light for exploring the parellels of the post-war."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:160]:
```suggestion
	desc = "A sturdy helmet with a built-in nuclear powered light for exploring post-war tunnels and caves."
```

File: mojave/items/clothing/head/helmet.dm [hunk 5]
```diff
@@ -120,74 +209,323 @@
...
 	flags_inv = HIDEEARS|HIDEFACE|HIDEHAIR
 
-/obj/item/clothing/head/helmet/ms13/utilitycap/deputy
-	name = "deputy's cap"
-	desc = "A formal refurbished pre-war cap. It's got a faded logo in the inner rim."
-	icon_state = "deputycap"
-	inhand_icon_state = "deputycap"
+/obj/item/clothing/head/hooded/ms13/assaultgas
+	name = "assault gas mask"
+	desc = "A US millitary assault mask, a relic of what man could have done."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:214]:
```suggestion
	desc = "A US military assault mask, a relic of what man could have done."
```

File: mojave/items/clothing/head/helmet.dm [hunk 6]
```diff
@@ -120,74 +209,323 @@
...
 
-/obj/item/clothing/head/helmet/ms13/cowboy/reinf
-	name = "reinforced cowboy hat"
-	desc = "A black cowboy hat reinforced with even more leather padding and a bit of metal, for good measure."
-	icon_state = "reinfcowboy"
-	inhand_icon_state = "reinfcowboy"
-	armor = list(melee = 35, bullet = 25, laser = 10, energy = 10, bomb = 20, bio = 10, rad = 10, fire = 10, acid = 10)
+/obj/item/clothing/head/hooded/ms13/plated
+	name = "plated cowl"
+	desc = "An armored cloth cowl, eyeholes poked poked through."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:230]:
```suggestion
	desc = "An armored cloth cowl, eyeholes poked through."
```

File: mojave/items/clothing/head/helmet.dm [hunk 7]
```diff
@@ -120,74 +209,323 @@
...
+/obj/item/clothing/head/hooded/ms13/cowl
+	name = "cowl"
+	desc = "A tattered cloth cowl."
+	icon_state = "cowl"
+	inhand_icon_state = "winterglovesbrown"
+	flags_inv = HIDEEARS|HIDEHAIR
+
+/obj/item/clothing/head/hooded/ms13/inquisitor
+	name = "inquisitors helm"
+	desc = "A menacing piece of religious headwear, pipes and gold shining throughout, must belong to someone of the highest intunement with god."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:245]:
```suggestion
	desc = "A menacing piece of religious headwear, pipes and gold shining throughout, must belong to someone of the highest attunement with god."
```

File: mojave/items/clothing/head/helmet.dm [hunk 8]
```diff
@@ -120,74 +209,323 @@
...
+	name = "tactical stealth hood"
+	desc = "A tactical pre-war black latex head and goggles, the read up display is long gone but this durable stealthwear is how you OPERATE."
+	icon_state = "tactical"
+	armor = list(melee = 20, bullet = 15, laser = 5, energy = 10, bomb = 10, bio = 5, rad = 5, fire = 10, acid = 5)
+	inhand_icon_state = "winterglovesblack"
+	flags_inv = HIDEEARS|HIDEHAIR
+
+/obj/item/clothing/head/hooded/ms13/hood
+	name = "green hood"
+	desc = "A simple handmade cloth hood, rather primative in its construction."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:284]:
```suggestion
	desc = "A simple handmade cloth hood, rather primitive in its construction."
```

File: mojave/items/clothing/head/helmet.dm [hunk 9]
```diff
@@ -120,74 +209,323 @@
...
+
+/obj/item/clothing/head/helmet/ms13/fedora/yellow
+	name = "fedora"
+	desc = "A bright and cheerful yellow fedora."
+	icon_state = "fedyellow"
+	inhand_icon_state = "sheriffhat"
+
+/obj/item/clothing/head/helmet/ms13/bowler
+	name = "bowler hat"
+	desc = "A classy gentlemans bowler hat."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:339]:
```suggestion
	desc = "A classy gentlemen bowler hat."
```

File: mojave/items/clothing/head/helmet.dm [hunk 10]
```diff
@@ -120,74 +209,323 @@
...
+	inhand_icon_state = "sheriffhat"
+	armor = list(melee = 15, bullet = 5, laser = 15, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 10, acid = 10)
+
+/obj/item/clothing/head/helmet/ms13/cowboy/black
+	icon_state = "cowboyblack"
+	inhand_icon_state = "sheriffhat"
+
+/obj/item/clothing/head/helmet/ms13/prospector
+	name = "prospector's hat"
+	desc = "A whimsical hat commonly worn by post-war prospectors, pickin the remnants of scrap and waste from the society of old."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:356]:
```suggestion
	desc = "A whimsical hat commonly worn by post-war prospectors, picking the remnants of scrap and waste from the society of old."
```

File: mojave/items/clothing/head/helmet.dm [hunk 11]
```diff
@@ -120,74 +209,323 @@
...
+	inhand_icon_state = "sheriffhat"
+	armor = list(melee = 15, bullet = 5, laser = 15, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 10, acid = 10)
+
+/obj/item/clothing/head/helmet/ms13/prospector/brown
+	icon_state = "prospectorbrown"
+	inhand_icon_state = "sheriffhat"
+
+/obj/item/clothing/head/helmet/ms13/millitia
+	name = "millitia hat"
+	desc = "A sturdy leather wide-brimmed millitia hat, the common mans garb of freedom."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:367]:
```suggestion
	desc = "A sturdy leather wide-brimmed militia hat, the common garb of free men."
```

File: mojave/items/clothing/head/helmet.dm [hunk 12]
```diff
@@ -120,74 +209,323 @@
...
+	name = "pirate hat"
+	desc = "Y'arr."
+	icon_state = "pirate"
+	inhand_icon_state = "fedora"
+
+// Tall Helmet //
+
+/obj/item/clothing/head/helmet/ms13/tall/regal
+	name = "fancy bicorn"
+	desc = "This fanciful hat belongs on a leader or in a museum, gold embroidered and navy silk emenate the barings of class and royalty far beyond those the modern common man perceives."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:388]:
```suggestion
	desc = "This fanciful hat belongs to a leader or in a museum. The gold and navy silk embroidery is classy and regal, conferring to those who wear it a bearing far beyond what is usually seen in the post-war wastelands."
```

File: mojave/items/clothing/head/helmet.dm [hunk 13]
```diff
@@ -120,74 +209,323 @@
...
 	inhand_icon_state = "general"
 
-/obj/item/clothing/head/helmet/ms13/cowboy/padded/detective
-	name = "detective's hat"
-	desc = "A wide brimmed hat normally worn by detectives. Padded for extra protection."
-	icon_state = "detective"
-	inhand_icon_state = "detective"
+/obj/item/clothing/head/helmet/ms13/utilitycap/deputy
+	name = "deputy's cap"
+	desc = "A formal refurbished pre-war cap. It's got a faded logo in the inner rim."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:422]:
```suggestion
	desc = "A formal refurbished pre-war cap. It's got a faded logo on the inner rim."
```

File: mojave/items/clothing/head/helmet.dm [hunk 14]
```diff
@@ -120,74 +209,323 @@
...
+	inhand_icon_state = "deputycap"
 
-/obj/item/clothing/head/helmet/ms13/cowboy/padded/detective/grey
-	name = "grey detective's hat"
-	desc = "A wide brimmed hat normally worn by detectives. This one is grey, but otherwise no different."
-	icon_state = "detectivealt"
-	inhand_icon_state = "detectivealt"
+/obj/item/clothing/head/helmet/ms13/captain
+	name = "captain's hat"
+	desc = "A pre-war millitary Captains hat, most likely used for land ranks. A true display of fashion and power all in one."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:428]:
```suggestion
	desc = "A pre-war military Captains hat, most likely used for land ranks. A true display of fashion and power all in one."
```

File: mojave/items/clothing/head/helmet.dm [hunk 15]
```diff
@@ -120,74 +209,323 @@
...
+	icon_state = "captain"
+	inhand_icon_state = "fedora"
+	armor = list(melee = 10, bullet = 5, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 10, acid = 10)
+
+/obj/item/clothing/head/helmet/ms13/captain/white
+	desc = "A pre-war millitary Captains hat, most likely used for sea ranks. A true display of aptitudue and power all in one."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:434]:
```suggestion
	desc = "A pre-war military Captains hat, most likely used for sea ranks. A true display of aptitude and power all in one."
```

File: mojave/items/clothing/head/helmet.dm [hunk 16]
```diff
@@ -120,74 +209,323 @@
...
+	icon_state = "captainwhite"
+	inhand_icon_state = "fedora"
+
+/obj/item/clothing/head/helmet/ms13/captain/commie
+	name = "kapitan's hat"
+	desc = "A pre-war strange millitary Captains hat, A strange font and almost moonrune-like lettering covers this one, clearly communism at its finest."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:440]:
```suggestion
	desc = "A pre-war strange military Captains hat, A strange font and almost moonrune-like lettering covers this one, clearly communism at its finest."
```

File: mojave/items/clothing/head/helmet.dm [hunk 17]
```diff
@@ -120,74 +209,323 @@
...
+/obj/item/clothing/head/helmet/ms13/police
+	name = "policemans hat"
+	desc = "A beat-cops sturdy protector, eye of the LAW."
+	icon_state = "policeman"
+	inhand_icon_state = "fedora"
+	armor = list(melee = 10, bullet = 5, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 10, acid = 10)
+
+/obj/item/clothing/head/helmet/ms13/fisherman
+	name = "fishermans hat"
+	desc = "A waterproof fishermans hat, for keeping the line taught and fish ready. Technobugs and bait not included."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:465]:
```suggestion
	desc = "A waterproof fisherman hat, for keeping the line taught and fish ready. Technobugs and bait not included."
```

File: mojave/items/clothing/head/helmet.dm [hunk 18]
```diff
@@ -120,74 +209,323 @@
...
+	icon_state = ""
+	inhand_icon_state = "fedora"
+
+/obj/item/clothing/head/helmet/ms13/bandanacap
+	name = "bandana cap"
+	desc = "A green piece of cloth tied around a yellow cap, some form of gang sign or sytlistic intent, who knows."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:471]:
```suggestion
	desc = "A green piece of cloth tied around a yellow cap, some form of gang sign or stylistic intent, who knows."
```

File: mojave/items/clothing/head/helmet.dm [hunk 19]
```diff
@@ -120,74 +209,323 @@
...
+	icon_state = "bandanacap"
+	inhand_icon_state = "fedora"
+
+/obj/item/clothing/head/helmet/ms13/millitary
+	name = "millitary hat"
+	desc = "A navy US pre-war millitary cap, in fairly good condition."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:477]:
```suggestion
	desc = "A navy US pre-war military cap, in fairly good condition."
```

File: mojave/items/clothing/head/helmet.dm [hunk 20]
```diff
@@ -120,74 +209,323 @@
...
+	icon_state = "millitary"
+	inhand_icon_state = "fedora"
+	armor = list(melee = 10, bullet = 5, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0, fire = 10, acid = 10)
+
+/obj/item/clothing/head/helmet/ms13/beret
+	name = "gray beret"
+	desc = "A formal gray beret. Most likely used in off-service millitary duties."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:484]:
```suggestion
	desc = "A formal gray beret. Most likely used in off-service military duties."
```

File: mojave/items/clothing/head/helmet.dm [hunk 21]
```diff
@@ -120,74 +209,323 @@
...
+	name = "wool beanie"
+	desc = "A snug tattered and well-worn green beanie."
+	icon_state = "beanie"
+	inhand_icon_state = "fedora"
+
+// Misc Hats //
+
+/obj/item/clothing/head/helmet/ms13/chef
+	name = "chefs hat"
+	desc = "A professional chefs hat, hopefully there are no rats inside."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:511]:
```suggestion
	desc = "A professional chef hat, hopefully there is no rat inside."
```

File: mojave/items/clothing/head/helmet.dm [hunk 22]
```diff
@@ -120,74 +209,323 @@
...
+	icon_state = "chef"
+	inhand_icon_state = "chefhat"
+
+/obj/item/clothing/head/helmet/ms13/ushanka
+	name = "ushanka hat"
+	desc = "Typical winter apparel, the aptitude of slavic engineering."
```
Comment (hyperioo) [inline mojave/items/clothing/head/helmet.dm:517]:
```suggestion
	desc = "Typical winter apparel, the epitome of slavic engineering."
```

File: mojave/items/clothing/masks/masks.dm [hunk 23]
```diff
@@ -3,6 +3,58 @@
...
 	icon = 'mojave/icons/objects/clothing/masks.dmi'
 	worn_icon = 'mojave/icons/mob/clothing/mask.dmi'
+	lefthand_file = 'mojave/icons/mob/inhands/clothing_lefthand.dmi'
+	righthand_file = 'mojave/icons/mob/inhands/clothing_righthand.dmi'
+
+// Bandanas //
+
+/obj/item/clothing/mask/ms13/bandana
+	name = "bandana"
+	desc = "A tactical and cool piece of face wrap, useful againt the dust, heat and those other summbitches that want your head."
```
Comment (hyperioo) [inline mojave/items/clothing/masks/masks.dm:13]:
```suggestion
	desc = "A tactical and cool piece of face wrap, useful againt the dust, heat and those other sumbitches that want your head."
```

**Review by hyperioo — COMMENTED** [review]
Here you go.

**Comment by TheRealScarHomie** [comment]
merge ready

**Pull request merged**